android/src/firebase/core/TitaniumFirebaseCoreModule.java

```diff
@@ -36,25 +41,78 @@ public TitaniumFirebaseCoreModule()
 			FirebaseOptions.Builder options = new FirebaseOptions.Builder();
-			if (param.containsKey("APIKey")) {
-				options.setApiKey(param.getString("APIKey"));
-			}
-			if (param.containsKey("databaseURL")) {
-				options.setDatabaseUrl(param.getString("databaseURL"));
-			}
-			if (param.containsKey("projectID")) {
-				options.setProjectId(param.getString("projectID"));
-			}
-			if (param.containsKey("storageBucket")) {
-				options.setStorageBucket(param.getString("storageBucket"));
-			}
-			if (param.containsKey("applicationID")) {
-				options.setApplicationId(param.getString("applicationID"));
-			}
-			if (param.containsKey("GCMSenderID")) {
-				options.setGcmSenderId(param.getString("GCMSenderID"));
+
+			if (param.containsKey("file")) {
+				// open file and parse it
+				try {
+					JSONObject json = new JSONObject(loadJSONFromAsset(param.getString("file")));
+					JSONObject projectInfo = json.getJSONObject("project_info");
+					String packageName = TiApplication.getAppCurrentActivity().getPackageName();
+
+					if (projectInfo.has("storage_bucket")) {
+						storageBucket = projectInfo.getString("storage_bucket");
+					}
+					if (projectInfo.has("firebase_url")) {
+						databaseURL = projectInfo.getString("firebase_url");
+					}
+					if (projectInfo.has("project_number")) {
+						GCMSenderID = projectInfo.getString("project_number");
+					}
+					if (projectInfo.has("project_id")) {
+						projectID = projectInfo.getString("project_id");
+					}
+					if (json.has("client")) {
+						JSONArray clients = json.getJSONArray("client");
+						for (int i = 0,len = clients.length(); i < len; i++) {
+							JSONObject client = clients.getJSONObject(i);
+							JSONObject clientInfo = client.getJSONObject("client_info");
+							String pName = clientInfo.getJSONObject("android_client_info").getString("package_name");
+							if (pName.equals(packageName)) {
+								applicationID = client.getJSONObject("client_info").getString("mobilesdk_app_id");
+								apiKey = client.getJSONArray("api_key").getJSONObject(0).getString("current_key");
+							}
+						}
+					}
+				} catch (JSONException e) {
+					Log.e(LCAT, "Error parsing file");
+				}
+			} else {
+				// use parameters
+				if (param.containsKey("APIKey")) {
+					apiKey = param.getString("APIKey");
+				}
+				if (param.containsKey("databaseURL")) {
+					databaseURL = param.getString("databaseURL");
+				}
+				if (param.containsKey("projectID")) {
+					projectID = param.getString("projectID");
+				}
+				if (param.containsKey("storageBucket")) {
+					storageBucket = param.getString("storageBucket");
+				}
+				if (param.containsKey("applicationID")) {
+					applicationID = param.getString("applicationID");
+				}
+				if (param.containsKey("GCMSenderID")) {
+					GCMSenderID = param.getString("GCMSenderID");
+				}
 			}
+
+			Log.d(LCAT, "Firebase Options: " + apiKey+"|"+databaseURL+"|"+projectID+"|"+storageBucket+"|"+applicationID+"|"+GCMSenderID);
```

Comment:
Do we still need this? I see a possible security issue with it.

Reply:
no, was just for debug purpose. Removed it